code/src/main/kotlin/com/expediagroup/sdk/core/http/MediaType.kt

```diff
@@ -58,107 +59,89 @@ class MediaType(
      * @return `true` if this media type includes the given media type, `false` otherwise.
      */
     fun includes(other: MediaType): Boolean {
-        if (this.type == "*") {
-            return true
-        } else if (this.type.equals(other.type, ignoreCase = true)) {
-            if (this.subtype == "*" || this.subtype.equals(other.subtype, ignoreCase = true)) {
-                return true
-            }
-        }
-        return false
-    }
+        val typeMatches = this.type == "*" || this.type.equals(other.type, ignoreCase = true)
+        val subtypeMatches = this.subtype == "*" || this.subtype.equals(other.subtype, ignoreCase = true)
 
-    /**
-     * Returns a copy of this `MediaType` with the given charset parameter.
-     *
-     * @param charset The charset to set.
-     * @return A new `MediaType` instance with the specified charset.
-     */
-    fun withCharset(charset: Charset): MediaType {
-        val newParameters = parameters.toMutableMap()
-        newParameters["charset"] = charset.name()
-        return MediaType(type, subtype, newParameters)
+        return typeMatches && subtypeMatches
     }
 
+    /**
+     * Returns the full representation of a standard media type consisting of type/subtype followed by all parameters, if any
+     * */
     override fun toString(): String {
-        val params = parameters.entries.joinToString(separator = ";") { (key, value) ->
+        val formattedParams = parameters.entries.joinToString(separator = ";") { (key, value) ->
             "$key=$value"
         }
-        return if (params.isNotEmpty()) "$type/$subtype;$params" else "$type/$subtype"
+        return if (formattedParams.isNotEmpty()) "$type/$subtype;$formattedParams" else "$type/$subtype"
     }
 
-    override fun equals(other: Any?): Boolean {
-        if (this === other) return true
-        if (other !is MediaType) return false
-
-        return type.equals(other.type, ignoreCase = true) &&
-                subtype.equals(other.subtype, ignoreCase = true) &&
-                parameters.entries.all { (key, value) ->
-                    other.parameters[key]?.equals(value, ignoreCase = true) == true
-                }
-    }
+    companion object {
+        /**
+         * Factory method for creating a MediaType.
+         */
+        @JvmStatic
+        @JvmOverloads
+        fun of(type: String, subtype: String, parameters: Map<String, String> = emptyMap()): MediaType {
+            require(type.isNotBlank()) { "Type must not be blank" }
+            require(subtype.isNotBlank()) { "Subtype must not be blank" }
+
+            if (type == "*" && subtype != "*") {
+                throw IllegalArgumentException("Invalid media type format: type=$type, subtype=$subtype")
+            }
```

Comment:
We can come back and do this later, but should we consider having `.parse()` delegate some logic to `.of()` to reduce some duplication?